service/qos/service_level_controller.cc

```diff
@@ -410,4 +410,16 @@ future<> service_level_controller::do_remove_service_level(sstring name, bool re
     return make_ready_future();
 }
 
+void service_level_controller::on_join_cluster(const gms::inet_address& endpoint, bool this_node) { }
+
+void service_level_controller::on_leave_cluster(const gms::inet_address& endpoint, bool this_node) {
```

Comment:
We can pass the ip address of this node to service_level_controller to avoid modifying this_node. It is weird and redundant.

Reply:
I disagree. I think that exposing the fact that nodes are identified entirely by their IP is redundant. What if we will support IP change one day?

Reply:
I could make `service::storage_service::get_broadcast_address()` public but it feels too intrusive for me.
If this is weird to you, I can add another subscription for local_endpoint_notifications. Do you think it is a better solution?

Reply:
@asias does my last solution proposal sounds better?

Reply:
There is no point to change the ip address of the node while decommissioning the node. We do not support that neither. 

IMO, It is over kill to add a another subscription for local_endpoint_notifications and it is unnecessary because we can know if the node is the node itself or not  using the endpoint parameter.

Reply:
When we support changing the ip address in the middle without restarting a node, we probably need to change on_leave_cluster(inet_address endpoint)  to on_leave_cluster(node_id id) instead. We can still use the node_id to tell if the node is is_node.

Reply:
OK @asias fair enough. I will change it.